web/src/components/dashboards/perses/PersesWrapper.tsx

```diff
@@ -229,52 +226,26 @@ export function PersesPrometheusDatasourceWrapper({
   children,
   dashboardResource,
 }: PersesPrometheusDatasourceWrapperProps) {
-  const csrfToken = useCookieWatcher('csrf-token', { valueOnly: true });
+  useCookieWatcher('csrf-token', { valueOnly: true });
   const { t } = useTranslation(process.env.I18N_NAMESPACE);
-  const datasourceApi = React.useMemo(() => {
-    return new CachedDatasourceAPI(new OcpDatasourceApi(csrfToken, t, PERSES_PROXY_BASE_PATH));
-  }, [csrfToken, t]);
-
-  return (
-    <DatasourceStoreProvider dashboardResource={dashboardResource} datasourceApi={datasourceApi}>
-      <DataQueriesProvider definitions={queries}>{children}</DataQueriesProvider>
-    </DatasourceStoreProvider>
+  const [csrfToken, setCsrfToken] = React.useState(getCSRFToken());
+  const [datasourceApi, setDatasourceApi] = React.useState(
```

Comment:
This probably defeats the purpose of caching the API, in principle the api should not change when the CSRF token changes. Wouldn't it be better to pass the `getCSRFToken` function instead of the actual CSRFToken, this way we can still cache the API but avoid caching the token.